packages/codemirror-lang-chordpro/src/syntax.grammar

```diff
@@ -1,40 +1,37 @@
-@top Program { expression* }
+@top Program { line* }
 
-@skip { whitespace }
+@skip { space }
 
-expression {
-  Chord |
-  MetaData | 
-  Bracket |
-  Comment |
-  String
-}
-
-@tokens {
-
-  whitespace { $[ \n\r\t] }
-
-  SymbolChord { ($[A-G] $[b|#]? ($[a-zA-Z0-9\(\#)]*)?) ( '/' $[A-G] $[b|#]?)? }
-
-  NumericChord { ($[b|#]? $[1-7] ($[a-zA-Z0-9\(\#)]*)?) ( '/' $[b|#]? $[1-7])?  }
-  
-  NumeralChord { ($[b|#]? ('I' | 'II' | 'III' | 'IV' | 'V' | 'VI' | 'VII') ($[a-zA-Z0-9\(\)#]*)?) ( '/' $[b|#]? ('I' | 'II' | 'III' | 'IV' | 'V' | 'VI' | 'VII'))?  }
+line { (Comment | Directive | Lyrics) (eol | eof) }
 
-  Chord { '[' ( SymbolChord | NumericChord | NumeralChord | $[x] | '|' | '/' ) ']' }
-  
-  Bracket { '[' | ']' }
+Directive { "{" DirectiveName (":" DirectiveValue)? "}" }
+Lyrics { ( "[" Chord "]" | Lyric)+ }
+Chord { symbolChord | numericChord | numeralChord | rhythmSymbol }
 
-  @precedence { Chord, SymbolChord, NumericChord, NumeralChord }
-
-  MetaData { "{" char* ":" char* "}" }
-
-  String { '"' char* '"' }
-  
-  char { $[\u{20}\u{21}\u{23}-\u{5b}\u{5d}-\u{10ffff}] | "\\" esc }
-  esc  { $["\\\/bfnrt] | "u" hex hex hex hex }
-  hex  { $[0-9a-fA-F] }
-
-  Comment { "//" ![\n]* }
+@tokens {
+  // Directives
+  DirectiveName { (@asciiLetter | "_")+ }
+  DirectiveValue { ![}\n]+ }
+  @precedence { DirectiveValue, space }
+
+  // Chords
+  symbolChord { ($[A-G] accidental? ($[a-zA-Z0-9\(\#)]*)?) ( '/' $[A-G] accidental?)? }
+  numericChord { (accidental? $[1-7] ($[a-zA-Z0-9\(\#)]*)?) ( '/' accidental? $[1-7])?  }
+  numeralChord { (accidental? ('I' | 'II' | 'III' | 'IV' | 'V' | 'VI' | 'VII') ($[a-zA-Z0-9\(\)#]*)?) ( '/' accidental? ('I' | 'II' | 'III' | 'IV' | 'V' | 'VI' | 'VII'))?  }
+  rhythmSymbol { $[x|/] }
+  accidental { $[b#] }
```

Comment:
I'm not sure this grammar needs to be strict about what goes between the `[` brackets `]` at this point. It might later once we get into linting, but I think we need to have a conversation about doing linting in the editor vs in ChordSheetJS.

Reply:
My thoughts were that if we are using this grammar in a chords over words sheet. It might be easier to differentiate between chords and lyrics. Factoring in chords only lines, and different ways of writing charts. 

While I defintiley don't disagree that I'm not sure how strict it should be. This was where my head was at starting off.

Reply:
👍  That makes sense.